Classes/Domain/ExtractionManager.php

```diff
@@ -140,4 +140,31 @@ public function extractMetaData(Asset $asset) : MetaDataCollection
 
         return $metaDataCollection;
     }
+
+    /**
+     * @param Asset $asset
+     * @return MetaDataCollection
+     * @throws ExtractorException
+     * @throws UnknownObjectException
+     */
+    public function onAssetRemoved(Asset $asset) : MetaDataCollection
+    {
+        if ($asset instanceof ImageVariant) {
+            $asset = $asset->getOriginalAsset();
+        }
+
+        $flowResource = $asset->getResource();
+        if ($flowResource === null) {
+            throw new ExtractorException('Resource of Asset "' . $asset->getTitle() . '"" not found.', 1484060541);
+        }
+
+        $metaDataCollection = new MetaDataCollection();
+        $this->buildAssetMetaData($asset, $metaDataCollection);
+
+        // Because the signal assetRemoved was emitted the deleted flag for $flowResource is true
+        // thus the signal metaDataCollectionUpdated finally deletes the metadata for the asset
+        $this->metaDataManager->emitMetaDataCollectionUpdated($asset, $metaDataCollection);
```

Comment:
This seems a bit too much magic/indirection. And emiting a signal from a different class (in a different package even) is a bit too much "knowledge about internals" at work. Isn't there a more explicit way?

Reply:
Right, I wanted to avoid a call to `mapMetaData` done by `updateMetaDataForAsset` from the MetadataManager. But you are right, the `mapMetaData` ignores delted resources, so I changed the call to `updateMetaDataForAsset()`.